custom/aaa/models.py

```diff
@@ -0,0 +1,272 @@
+    # need to investigate that there is not an app workflow or error state
+    # where a person has multiple child_health cases
+    person_case_id = models.TextField(unique=True)
+    child_health_case_id = models.TextField(primary_key=True)
+
+    # For now, these should be the same between child_health and person case
+    # but going with child_health as there's a potential for person case to live
+    # on after the child_health case stops being tracked (into adolescent girl, or woman)
+    opened_on = models.DateField(help_text="child_health.opened_on")
+    closed_on = models.DateField(help_text="child_health.opened_on", null=True)
+
+    dob = models.DateField(null=True)
+    sex = models.TextField(null=True)
+    migration_status = models.TextField(null=True)
+
+    @classmethod
+    def agg_from_child_health_case_ucr(cls, domain, window_start, window_end):
+        doc_id = StaticDataSourceConfiguration.get_doc_id(domain, 'reach-child_health_cases')
+        config, _ = get_datasource_config(doc_id, domain)
+        ucr_tablename = get_table_name(domain, config.table_id)
+
+        return """
+        INSERT INTO "{child_tablename}" AS child (
+            domain, person_case_id, child_health_case_id, opened_on, closed_on
+        ) (
+            SELECT
+                %(domain)s AS domain,
+                child_health.person_case_id AS person_case_id,
+                child_health.doc_id AS child_health_case_id,
+                child_health.opened_on AS opened_on,
+                child_health.closed_on AS closed_on
+            FROM "{child_health_cases_ucr_tablename}" child_health
+            WHERE inserted_at >= %(window_start)s AND inserted_at < %(window_end)s
+        )
+        ON CONFLICT (person_case_id) DO UPDATE SET
+           closed_on = EXCLUDED.closed_on
+        """.format(
+            child_tablename=cls._meta.db_table,
+            child_health_cases_ucr_tablename=ucr_tablename,
+        ), {'domain': domain, 'window_start': window_start, 'window_end': window_end}
+
+    @classmethod
+    def agg_from_person_case_ucr(cls, domain, window_start, window_end):
+        doc_id = StaticDataSourceConfiguration.get_doc_id(domain, 'reach-person_cases')
+        config, _ = get_datasource_config(doc_id, domain)
+        ucr_tablename = get_table_name(domain, config.table_id)
+
+        return """
+        UPDATE "{child_tablename}" AS child SET
+            household_case_id = person.household_case_id,
+            dob = person.dob,
+            sex = person.sex,
+            migration_status = person.migration_status
+        FROM (
+            SELECT
+                household_case_id,
+                doc_id,
+                dob,
+                sex,
+                migration_status
+            FROM "{person_cases_ucr_tablename}"
+            WHERE inserted_at >= %(window_start)s AND inserted_at < %(window_end)s
+        ) person
+        WHERE child.person_case_id = person.doc_id
```

Comment:
Is there a case where the row may not exist in the table yet?

Reply:
There will be many  person cases that will not exist in the child table.  there does exist a case where:

1) person case is updated
2) aggregation window happens
3) child case updated
4) aggregation window happens
5) person case is not in this inserted_at window.

I think this could be solved in the background with a task that monitors the child table to find records that have these key  columns NULL, or perhaps a two step aggregation where we update person cases that have changed and also update records with NULL columns. I may change this for the pilot to do full data aggregations in the short term while this is more solved